src/main/java/com/google/devtools/build/lib/remote/AbstractRemoteActionCache.java

```diff
@@ -265,7 +265,9 @@ public void onFailure(Throwable t) {
           execRoot.getRelative(file.getPath()).delete();
         }
         for (OutputDirectory directory : result.getOutputDirectoriesList()) {
-          FileSystemUtils.deleteTree(execRoot.getRelative(directory.getPath()));
+          // Only delete the directories below the output directories because the output
+          // directories will not be re-created
+          FileSystemUtils.deleteTreesBelow(execRoot.getRelative(directory.getPath()));
```

Comment:
just had a thought, this method's name and documentation implies that it deletes only the subdirectories inside the given directory, but there may also be regular files rooted at this directory. @buchgr can you confirm that it deletes the files as well? I traced through the code, but wasn't sure.

Reply:
Would there ever be regular files that weren't part of the output? The output ones are deleted above, but otherwise I can stop using this helper and directly loop over the children of this directory

Reply:
Another option would be to leave the original line, and call `createDirectoryAndParents` afterwards